CPAC/pipeline/utils.py

```diff
@@ -65,15 +66,13 @@ def name_fork(resource_idx, cfg, json_info, out_dct):
                                                      'filt', filt_value)
     if cfg.switch_is_on(['nuisance_corrections',
                          '2-nuisance_regression', 'run']):
-        reg_value = None
-        if ('regressors' in json_info.get('CpacVariant', {})
-                and json_info['CpacVariant']['regressors']):
-            reg_value = json_info['CpacVariant'][
-                'regressors'
-            ][0].replace('nuisance_regressors_generation_', '')
-        elif cfg.switch_is_off(['nuisance_corrections',
-                                '2-nuisance_regression', 'run']):
-            reg_value = 'Off'
+        variants = [variant.split('_')[-1] for variant in
+                    chain.from_iterable(json_info['CpacVariant'].values()) if
+                    variant.startswith('nuisance_regressors_generation')]
+        if cfg.switch_is_off(['nuisance_corrections', '2-nuisance_regression',
+                              'run']):
+            variants.append['Off']
+        reg_value = variants[0] if variants else None
```

Comment:
I think these tweaks
```suggestion
                    chain.from_iterable(json_info.get('CpacVariant', {}).values()) if
                    variant.startswith('nuisance_regressors_generation')]
        if cfg.switch_is_off(['nuisance_corrections', '2-nuisance_regression',
                              'run']):
            variants.append('Off')
        reg_value = variants[0] if variants else None
```
should make this function more robust and get the smoke tests passing again

Reply:
Looks good, let's do it!